### PR TITLE
Remove legacy context access in ParentNewerCondition

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Remove access to `legacy_context` and move the functionality to `NonAGVInstanceInterface`

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by schrockn** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21924?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#21924** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21924?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#21923** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21923?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21922** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21922?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21921** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21921?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21919** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21919?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21918** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21918?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21917** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21917?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @schrockn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_context.py [hunk 1]
```diff
@@ -43,6 +46,21 @@ def get_asset_subset_updated_after_time(
             asset_key=asset_key, after_time=after_time
         )
 
+    def get_parent_asset_partitions_updated_after_child(
+        self,
+        *,
+        asset_partition: AssetKeyPartitionKey,
+        parent_asset_partitions: AbstractSet[AssetKeyPartitionKey],
+        respect_materialization_data_versions: bool,
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_context.py:54]: we can take off the respect_materialization_data_versions flag here -- we'll want to remove it entirely and so we can just set it to False when calling the queryer method

**Review by OwenKephart — APPROVED** [review]
note comment, otherwise looks good

**Comment by schrockn** [comment]
### Merge activity

* **May 17, 12:31 PM EDT**: @schrockn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/21924).

**Pull request merged**